pkg/querier/querier.go

```diff
@@ -80,24 +80,26 @@ type ChunkStore interface {
 	Get(ctx context.Context, userID string, from, through model.Time, matchers ...*labels.Matcher) ([]chunk.Chunk, error)
 }
 
-// New builds a queryable and promql engine.
-func New(cfg Config, distributor Distributor, chunkStore ChunkStore) (storage.Queryable, *promql.Engine) {
-	iteratorFunc := mergeChunks
+func getChunksIteratorFunction(cfg Config) chunkIteratorFunc {
 	if cfg.BatchIterators {
-		iteratorFunc = batch.NewChunkMergeIterator
+		return batch.NewChunkMergeIterator
 	} else if cfg.Iterators {
-		iteratorFunc = iterators.NewChunkMergeIterator
+		return iterators.NewChunkMergeIterator
 	}
+	return mergeChunks
+}
+
+func NewChunkStoreQueryable(cfg Config, chunkStore ChunkStore) storage.Queryable {
+	return newChunkStoreQueryable(chunkStore, getChunksIteratorFunction(cfg))
+}
+
+// New builds a queryable and promql engine.
+func New(cfg Config, distributor Distributor, storeQueryable storage.Queryable) (storage.Queryable, *promql.Engine) {
+	iteratorFunc := getChunksIteratorFunction(cfg)
 
 	var queryable storage.Queryable
-	if cfg.IngesterStreaming {
-		dq := newIngesterStreamingQueryable(distributor, iteratorFunc)
-		queryable = newUnifiedChunkQueryable(dq, chunkStore, distributor, iteratorFunc, cfg)
-	} else {
-		cq := newChunkStoreQueryable(chunkStore, iteratorFunc)
-		dq := newDistributorQueryable(distributor)
-		queryable = NewQueryable(dq, cq, distributor, cfg)
-	}
+	dq := newDistributorQueryable(distributor, cfg.IngesterStreaming, iteratorFunc)
```

Comment:
What do you thing if we rename `dq` to `distributorQueryable` for clarity, even if longer?